ignite/engine/__init__.py

```diff
@@ -69,48 +302,71 @@ def create_supervised_trainer(
+            `torch.cuda.amp <https://pytorch.org/docs/stable/amp.html>`_ for ``amp`` and
+            using `apex <https://nvidia.github.io/apex>`_ for ``apex``. (default: None)
+        scaler (torch.cuda.amp.GradScaler, bool, optional): GradScaler instance for gradient scaling if `torch>=1.6.0`
+            and ``amp_mode`` is ``amp``. If ``amp_mode`` is ``apex``, this argument will be ignored.
+            If True, will create default GradScaler. If GradScaler instance is passed, it will be used instead.
+            (default: False)
+
+    Note:
+        If ``scaler`` is True, GradScaler instance will be created internally and trainer state has attribute named
+        ``scaler`` for that instance and can be used for saving and loading.
 
     Note:
         `engine.state.output` for this engine is defined by `output_transform` parameter and is the loss
         of the processed batch by default.
 
     .. warning::
-
         The internal use of `device` has changed.
         `device` will now *only* be used to move the input data to the correct device.
         The `model` should be moved by the user before creating an optimizer.
         For more information see:
 
         - `PyTorch Documentation <https://pytorch.org/docs/stable/optim.html#constructing-it>`_
-
         - `PyTorch's Explanation <https://github.com/pytorch/pytorch/issues/7844#issuecomment-503713840>`_
 
+    .. warning::
+        If ``amp_mode='apex'`` , the model(s) and optimizer(s) must be initialized beforehand
+        since ``amp.initialize`` should be called after you have finished constructing your model(s)
+        and optimizer(s), but before you send your model through any DistributedDataParallel wrapper.
+
+        See more: https://nvidia.github.io/apex/amp.html#module-apex.amp
+
     Returns:
         Engine: a trainer engine with supervised update function.
+
+    .. versionchanged:: 0.5.0
+
+        - Added ``amp_mode`` argument for automatic mixed precision.
+        - Added ``scaler`` argument for gradient scaling.
     """
 
     device_type = device.type if isinstance(device, torch.device) else device
     on_tpu = "xla" in device_type if device_type is not None else False
-
-    if on_tpu and not idist.has_xla_support:
-        raise RuntimeError("In order to run on TPU, please install PyTorch XLA")
-
-    def _update(engine: Engine, batch: Sequence[torch.Tensor]) -> Union[Any, Tuple[torch.Tensor]]:
-        model.train()
-        optimizer.zero_grad()
-        x, y = prepare_batch(batch, device=device, non_blocking=non_blocking)
-        y_pred = model(x)
-        loss = loss_fn(y_pred, y)
-        loss.backward()
-
-        if on_tpu:
-            xm.optimizer_step(optimizer, barrier=True)
-        else:
-            optimizer.step()
-
-        return output_transform(x, y, y_pred, loss)
+    mode, scaler_ = _check_arg(on_tpu, amp_mode, scaler)
```

Comment:
```suggestion
    mode, _scaler = _check_arg(on_tpu, amp_mode, scaler)
```
Please prefix local variables / function by `_` rather than suffix.